src/app/dl/study-code/[jobId]/[fileName]/route.ts

```diff
@@ -9,14 +10,13 @@ export const GET = async (_: Request, { params }: { params: Promise<{ jobId: str
         return NextResponse.json({ error: 'no job id or file name provided' }, { status: 400 })
     }
 
-    const canUserAccessJob = await checkUserAllowedJobView(jobId)
+    const job = await jobInfoForJobId(jobId)
 
-    if (!canUserAccessJob) {
+    const session = await sessionFromClerk()
+    if (!session?.can('read', 'StudyJob')) {
```

Comment:
This is a general check for the ability to read any StudyJob.

However
https://github.com/safeinsights/management-app/blob/facb5277633c987b8c3a232583ba542fec9d78a9/src/lib/permissions.ts#L58 
defines the permission conditionally with a study org id.

Does not the check requires a specific StudyJob subject with an orgId to be evaluated correctly?

Reply:
yes it should use the more specific version, updated. ty!